### ISSUE TITLE
Fix the keras_tuner bug

### ISSUE THREAD

**Issue by GoatedChopin** [GoatedChopin/XRayBot#2]
https://stackoverflow.com/questions/55731589/cnn-keras-valueerror-negative-dimension-size-caused-by-subtracting-3-from-2-fo

It seems like maybe the padding is the issue, maybe you can add padding = "same" on each Conv2D layer to resolve the issue.

ex. _model.add(layers.Conv2D(64, (3, 3), activation='relu', padding='same'))_

**Comment by GoatedChopin** [comment]
https://stackoverflow.com/questions/47324571/keras-valueerror-negative-dimension-size-caused-by-subtracting-5-from-1

When you try to add a pooling layer larger than the filters that it's pooling from, you get a negative dimension error. I could try reducing the size of the model (less layers), reducing the size of the pools, or reducing the number of pools. Any of these options should work.

A more permanent solution would be to specify that the pool size is selected according to the output of the previous layer, but I'm not sure how to do that.

**Comment by GoatedChopin** [comment]
https://github.com/keras-team/keras/issues/1981#:~:text=Using%20the._keras_shapeattribute%20of%20the%20output%20of%20any%20instantiated,%3D%20keras.layers.Input%28x_train.shape%5B1%3A%5D%29%20conv1%20%3D%20keras.layers.Conv1D%2816%2C%208%2C%20padding%3D%27same%27%29%28x%29%20print%28conv1._keras_shape%29

It would seem that creating a layer by assigning it to a variable allows you to test the output shape using _layer._keras_shape_
In a future iteration, the program could still sample random pooling sizes and simply use the shape of the previous layer as the upper bound of the random sampling. This way, there's plenty of variety without the danger of sampling an impossible number.

x = keras.layers.Input(x_train.shape[1:])
conv1 = keras.layers.Conv1D(16, 8, padding='same')(x)
print(conv1._keras_shape)

**Comment by GoatedChopin** [comment]
Maybe try using the HyperRestNet hypermodel in keras_tuner.applications
from keras_tuner.applications.resnet import HyperResNet

Pass this HyperResNet model into the keras_tuner object to search instead of the dumpster fire you built. It'll probably work way better lmao.

https://blog.tensorflow.org/2020/01/hyperparameter-tuning-with-keras-tuner.html
